internal/service/alertconfiguration/model_alert_configuration_test.go

```diff
@@ -0,0 +1,557 @@
+	expectedTFModel     []alertconfiguration.TfMatcherModel
+}
+
+type sdkToTFThresholdConfigModelTestCase struct {
+	name                        string
+	SDKResp                     *admin.GreaterThanRawThreshold
+	currentStateThresholdConfig []alertconfiguration.TfThresholdConfigModel
+	expectedTFModel             []alertconfiguration.TfThresholdConfigModel
+}
+
+type sdkToTFAlertConfigurationModelTestCase struct {
+	name                           string
+	SDKResp                        *admin.GroupAlertsConfig
+	currentStateAlertConfiguration *alertconfiguration.TfAlertConfigurationRSModel
+	expectedTFModel                alertconfiguration.TfAlertConfigurationRSModel
+}
+
+const (
+	group               string  = "GROUP"
+	previousIntervalMin int     = 10
+	intervalMin         int     = 5
+	delayMin            int     = 0
+	enabled             bool    = true
+	disabled            bool    = false
+	previousOperator    string  = "MORE_THAN"
+	operator            string  = "LESS_THAN"
+	threshold           float64 = 99.0
+	units               string  = "RAW"
+	mode                string  = "AVERAGE"
+)
+
+func TestNotificationSDKToTFModel(t *testing.T) {
+	testCases := []sdkToTFNotificationModelTestCase{
+		{
+			name: "Complete SDK response",
+			SDKResp: &[]admin.AlertsNotificationRootForGroup{
+				{
+					TypeName:     admin.PtrString(group),
+					IntervalMin:  admin.PtrInt(intervalMin),
+					DelayMin:     admin.PtrInt(delayMin),
+					SmsEnabled:   admin.PtrBool(disabled),
+					EmailEnabled: admin.PtrBool(enabled),
+					Roles: []string{
+						"GROUP_DATA_ACCESS_READ_ONLY",
+						"GROUP_CLUSTER_MANAGER",
+						"GROUP_DATA_ACCESS_ADMIN",
+					},
+				},
+			},
+			currentStateNotifications: []alertconfiguration.TfNotificationModel{
+				{
+					TypeName:     types.StringValue(group),
+					IntervalMin:  types.Int64Value(int64(previousIntervalMin)),
+					DelayMin:     types.Int64Value(int64(delayMin)),
+					SMSEnabled:   types.BoolValue(disabled),
+					EmailEnabled: types.BoolValue(enabled),
+					Roles: []string{
+						"GROUP_DATA_ACCESS_READ_ONLY",
+						"GROUP_CLUSTER_MANAGER",
+						"GROUP_DATA_ACCESS_ADMIN",
+					},
+				},
+			},
+			expectedTFModel: []alertconfiguration.TfNotificationModel{
+				{
+					TypeName:     types.StringValue(group),
+					IntervalMin:  types.Int64Value(int64(intervalMin)),
+					DelayMin:     types.Int64Value(int64(delayMin)),
+					SMSEnabled:   types.BoolValue(disabled),
+					EmailEnabled: types.BoolValue(enabled),
+					Roles: []string{
+						"GROUP_DATA_ACCESS_READ_ONLY",
+						"GROUP_CLUSTER_MANAGER",
+						"GROUP_DATA_ACCESS_ADMIN",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel := alertconfiguration.NewTFNotificationModelList(*tc.SDKResp, tc.currentStateNotifications)
+			if !reflect.DeepEqual(resultModel, tc.expectedTFModel) {
+				t.Errorf("created terraform model did not match expected output")
```

Comment:
it'd be better to show expected and result in error message so fail tests can be understood, same for other tests

Reply:
Done!